### PR TITLE
Fixes to simplify the use of the repo for users external to K-Scale Labs

### PR DESCRIPTION
**Pull request by traversaro**
Hello, thanks for open sourcing the repo! I tried to run it locally but I faced some problems, so I did some fixes, in particular:

* Use relative path in .gitmodules to support users without ssh configured  (https://stackoverflow.com/a/48067316/1379427)
* Permit to avoid to use private repo kscalelabs/isaacgym.git and rely on local isaacgym
* Fix use of `&` instead of `&&` in command to download stompy model
* Fix name of `sim/scripts/create_fixed_urdf.py` script

I did not actually tested the training, so I do not know if everything works properly (as I do not know if kscalelabs/isaacgym.git has local changes), but at least the installation completes correctly.

<details>

<summary> `conda list` output for reproducibility </summary>

~~~
(kscale-sim-library) traversaro@IITBMP014LW012:~/kscale-sim$ conda list
# packages in environment at /home/traversaro/miniforge3/envs/kscale-sim-library:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
absl-py                   2.1.0                    pypi_0    pypi
appdirs                   1.4.4                    pypi_0    pypi
black                     24.4.0                   pypi_0    pypi
bzip2                     1.0.8                hd590300_5    conda-forge
ca-certificates           2024.2.2             hbcca054_0    conda-forge
cachetools                5.3.3                    pypi_0    pypi
certifi                   2024.2.2                 pypi_0    pypi
charset-normalizer        3.3.2                    pypi_0    pypi
click                     8.1.7                    pypi_0    pypi
contourpy                 1.1.1                    pypi_0    pypi
cycler                    0.12.1                   pypi_0    pypi
darglint                  1.8.1                    pypi_0    pypi
docker-pycreds            0.4.0                    pypi_0    pypi
exceptiongroup            1.2.0                    pypi_0    pypi
filelock                  3.13.4                   pypi_0    pypi
fonttools                 4.51.0                   pypi_0    pypi
fsspec                    2024.3.1                 pypi_0    pypi
gitdb                     4.0.11                   pypi_0    pypi
gitpython                 3.1.43                   pypi_0    pypi
glfw                      2.7.0                    pypi_0    pypi
google-auth               2.29.0                   pypi_0    pypi
google-auth-oauthlib      1.0.0                    pypi_0    pypi
grpcio                    1.62.1                   pypi_0    pypi
humanoid                  1.0.0                     dev_0    <develop>
idna                      3.7                      pypi_0    pypi
imageio                   2.34.0                   pypi_0    pypi
importlib-metadata        7.1.0                    pypi_0    pypi
importlib-resources       6.4.0                    pypi_0    pypi
iniconfig                 2.0.0                    pypi_0    pypi
isaacgym                  1.0rc4                    dev_0    <develop>
jinja2                    3.1.3                    pypi_0    pypi
kiwisolver                1.4.5                    pypi_0    pypi
kscale-sim-library        0.0.1                    pypi_0    pypi
ld_impl_linux-64          2.40                 h41732ed_0    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 13.2.0               h807b86a_5    conda-forge
libgomp                   13.2.0               h807b86a_5    conda-forge
libnsl                    2.0.1                hd590300_0    conda-forge
libsqlite                 3.45.2               h2797004_0    conda-forge
libuuid                   2.38.1               h0b41bf4_0    conda-forge
libxcrypt                 4.4.36               hd590300_1    conda-forge
libzlib                   1.2.13               hd590300_5    conda-forge
markdown                  3.6                      pypi_0    pypi
markupsafe                2.1.5                    pypi_0    pypi
matplotlib                3.7.5                    pypi_0    pypi
mpmath                    1.3.0                    pypi_0    pypi
mujoco                    2.3.6                    pypi_0    pypi
mujoco-python-viewer      0.1.4                    pypi_0    pypi
mypy                      1.9.0                    pypi_0    pypi
mypy-extensions           1.0.0                    pypi_0    pypi
ncurses                   6.4.20240210         h59595ed_0    conda-forge
networkx                  3.1                      pypi_0    pypi
ninja                     1.11.1.1                 pypi_0    pypi
numpy                     1.23.5                   pypi_0    pypi
nvidia-cublas-cu12        12.1.3.1                 pypi_0    pypi
nvidia-cuda-cupti-cu12    12.1.105                 pypi_0    pypi
nvidia-cuda-nvrtc-cu12    12.1.105                 pypi_0    pypi
nvidia-cuda-runtime-cu12  12.1.105                 pypi_0    pypi
nvidia-cudnn-cu12         8.9.2.26                 pypi_0    pypi
nvidia-cufft-cu12         11.0.2.54                pypi_0    pypi
nvidia-curand-cu12        10.3.2.106               pypi_0    pypi
nvidia-cusolver-cu12      11.4.5.107               pypi_0    pypi
nvidia-cusparse-cu12      12.1.0.106               pypi_0    pypi
nvidia-nccl-cu12          2.19.3                   pypi_0    pypi
nvidia-nvjitlink-cu12     12.4.127                 pypi_0    pypi
nvidia-nvtx-cu12          12.1.105                 pypi_0    pypi
oauthlib                  3.2.2                    pypi_0    pypi
opencv-python             4.9.0.80                 pypi_0    pypi
openssl                   3.2.1                hd590300_1    conda-forge
packaging                 24.0                     pypi_0    pypi
pathspec                  0.12.1                   pypi_0    pypi
pillow                    10.3.0                   pypi_0    pypi
pip                       24.0               pyhd8ed1ab_0    conda-forge
platformdirs              4.2.0                    pypi_0    pypi
pluggy                    1.4.0                    pypi_0    pypi
protobuf                  4.25.3                   pypi_0    pypi
psutil                    5.9.8                    pypi_0    pypi
pyasn1                    0.6.0                    pypi_0    pypi
pyasn1-modules            0.4.0                    pypi_0    pypi
pyopengl                  3.1.7                    pypi_0    pypi
pyparsing                 3.1.2                    pypi_0    pypi
pytest                    8.1.1                    pypi_0    pypi
python                    3.8.19          hd12c33a_0_cpython    conda-forge
python-dateutil           2.9.0.post0              pypi_0    pypi
pyyaml                    6.0.1                    pypi_0    pypi
readline                  8.2                  h8228510_1    conda-forge
requests                  2.31.0                   pypi_0    pypi
requests-oauthlib         2.0.0                    pypi_0    pypi
rsa                       4.9                      pypi_0    pypi
ruff                      0.3.7                    pypi_0    pypi
scipy                     1.10.1                   pypi_0    pypi
sentry-sdk                1.45.0                   pypi_0    pypi
setproctitle              1.3.3                    pypi_0    pypi
setuptools                69.2.0             pyhd8ed1ab_0    conda-forge
six                       1.16.0                   pypi_0    pypi
smmap                     5.0.1                    pypi_0    pypi
sympy                     1.12                     pypi_0    pypi
tensorboard               2.14.0                   pypi_0    pypi
tensorboard-data-server   0.7.2                    pypi_0    pypi
tk                        8.6.13          noxft_h4845f30_101    conda-forge
tomli                     2.0.1                    pypi_0    pypi
torch                     2.2.2                    pypi_0    pypi
torchvision               0.17.2                   pypi_0    pypi
tqdm                      4.66.2                   pypi_0    pypi
triton                    2.2.0                    pypi_0    pypi
typing-extensions         4.11.0                   pypi_0    pypi
urllib3                   2.2.1                    pypi_0    pypi
wandb                     0.16.6                   pypi_0    pypi
werkzeug                  3.0.2                    pypi_0    pypi
wheel                     0.43.0             pyhd8ed1ab_1    conda-forge
xz                        5.2.6                h166bdaf_0    conda-forge
zipp                      3.18.1                   pypi_0    pypi
~~~

</details>

### CODE REVIEW

**Comment by codekansas** [comment]
Yeah, thanks for reviewing :) We can't make the IsaacGym repo public because it's proprietary so yea, you have to download it yourself. Great PR, thanks

File: README.md [hunk 1]
```diff
@@ -26,15 +26,26 @@ make install-dev
 ```
 
 3. Install third-party dependencies:
+
```
Comment (budzianowski) [inline README.md:29]: You can delete this paragraph and we can just keep the `make install-third-party-external`

**Review by budzianowski — APPROVED** [review]
Thanks for the fix! Just one nit.

File: README.md (hunk 1 above)
Comment (traversaro) [reply README.md:29]: Ok, done in https://github.com/kscalelabs/sim/commit/29a8d0e551312b089ee8447fc7cfb300fa031698 .

**Pull request merged**